src/index.c

```diff
@@ -1306,7 +1307,14 @@ static int NI_ReadSorted(void *ctx, RSIndexResult **hit) {
     if (nc->child->Read(nc->child->ctx, &cr) == INDEXREAD_EOF) {
       break;
     }
+
+    // Check for timeout with low granularity (MOD-5512)
+    if (TimedOut_WithCtx_Gran(&nc->timeoutCtx, 5000)) {
+      IITER_SET_EOF(&nc->base);
```

Comment:
We cannot test this at the moment

Reply:
Test added

Reply:
Completion: Add a heavier test to make sure we timeout here (in the `Not` iterator)